fastly/resource_fastly_service_authorization.go

```diff
@@ -0,0 +1,115 @@
+package fastly
+
+import (
+	"context"
+
+	gofastly "github.com/fastly/go-fastly/v6/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceServiceAuthorization() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceServiceAuthorizationCreate,
+		ReadContext:   resourceServiceAuthorizationRead,
+		UpdateContext: resourceServiceAuthorizationUpdate,
+		DeleteContext: resourceServiceAuthorizationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of this service authorization.",
+			},
+			"permission": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Default:          "full",
+				Description:      "The permissions to grant the user. Can be `full` (the default), `read_only`, `purge_select` or `purge_all`.",
+				ValidateDiagFunc: validateServiceAuthorizationPermission(),
+			},
+			"service_id": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: "The ID of the service to grant permissions for.",
+			},
+
+			"user_id": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: "The ID of the user which will receive the granted permissions.",
+			},
+		},
+	}
+}
+
+func resourceServiceAuthorizationCreate(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*FastlyClient).conn
+
+	sa, err := conn.CreateServiceAuthorization(&gofastly.CreateServiceAuthorizationInput{
+		ServiceID:  d.Get("service_id").(string),
+		UserID:     d.Get("user_id").(string),
+		Permission: gofastly.ServiceAuthorizationPermission(d.Get("permission").(string)),
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(sa.ID)
+
+	return nil
+}
+
+func resourceServiceAuthorizationRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*FastlyClient).conn
+
+	sa, err := conn.GetServiceAuthorization(&gofastly.GetServiceAuthorizationInput{
+		ID: d.Id(),
+	})
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(sa.ID)
+	d.Set("service_id", sa.ServiceID)
+	d.Set("user_id", sa.UserID)
+	d.Set("permission", sa.Permission)
+
+	return nil
+}
+
+func resourceServiceAuthorizationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*FastlyClient).conn
+
+	// Update Name and/or Role.
```

Comment:
Not sure what this line refers to? 

I think it's a side-effect of copy/pasting from another file (`fastly/resource_fastly_user.go`?)

Reply:
That's exactly it 😅  I'll sort it.